### PR TITLE
Two bug fixes in testcases

### PR DESCRIPTION
**Pull request by jschmidb**
These are two fixes for pre-z15 processors. One is an old, known problem, the other one is a bug in a newly added testcase for protected key support.

### CODE REVIEW

File: testcases/crypto/ec_func.c [hunk 1]
```diff
@@ -1387,7 +1387,8 @@ CK_RV run_GenerateECCKeyPairSignVerify()
         if (rc != CKR_OK) {
             if (rc == CKR_MECHANISM_PARAM_INVALID ||
                 rc == CKR_ATTRIBUTE_VALUE_INVALID ||
-                rc == CKR_CURVE_NOT_SUPPORTED) {
+                rc == CKR_CURVE_NOT_SUPPORTED ||
+                rc == CKR_FUNCTION_CANCELED) {
```
Comment (ifranzki) [inline testcases/crypto/ec_func.c:1391]: please see my mail about how to fix the CKR_FUNCTION_CANCELED case.

File: usr/lib/ep11_stdll/ep11_specific.c [hunk 2]
```diff
@@ -5897,17 +5870,24 @@ static CK_RV rsa_ec_generate_keypair(STDLL_TokData_t * tokdata,
...
+        if (!ep11tok_ec_curve_supported2(tokdata, publ_tmpl, &curve)) {
+            TRACE_ERROR("Curve not supported.\n");
+            return CKR_CURVE_NOT_SUPPORTED;
+        }
+    }
 
-    rc = build_ep11_attrs(tokdata, publ_tmpl, &new_publ_attrs, &new_publ_attrs_len, ktype, CKO_PUBLIC_KEY, curve_type);
+    rc = build_ep11_attrs(tokdata, publ_tmpl,
+                          &new_publ_attrs, &new_publ_attrs_len,
+                          ktype, CKO_PUBLIC_KEY, curve->curve_type);
```
Comment (ifranzki) [inline usr/lib/ep11_stdll/ep11_specific.c:5882]: what if variable curve is NULL here (e.g. non EC key)? 
Better use something like 
`curve != NULL ? curve->curve_type : -1`

File: usr/lib/ep11_stdll/ep11_specific.c [hunk 3]
```diff
@@ -5897,17 +5870,24 @@ static CK_RV rsa_ec_generate_keypair(STDLL_TokData_t * tokdata,
...
     if (rc != CKR_OK) {
         TRACE_ERROR("%s build_ep11_attrs failed with rc=0x%lx\n", __func__, rc);
         goto error;
     }
 
-    rc = build_ep11_attrs(tokdata, priv_tmpl, &new_priv_attrs, &new_priv_attrs_len, ktype, CKO_PRIVATE_KEY, curve_type);
+    rc = build_ep11_attrs(tokdata, priv_tmpl,
+                          &new_priv_attrs, &new_priv_attrs_len,
+                          ktype, CKO_PRIVATE_KEY, curve->curve_type);
```
Comment (ifranzki) [inline usr/lib/ep11_stdll/ep11_specific.c:5890]: same here

File: usr/lib/ep11_stdll/ep11_specific.c [hunk 4]
```diff
@@ -5916,7 +5896,7 @@ static CK_RV rsa_ec_generate_keypair(STDLL_TokData_t * tokdata,
     rc = check_key_attributes(tokdata, ktype, CKO_PUBLIC_KEY,
                               new_publ_attrs, new_publ_attrs_len,
                               &new_publ_attrs2, &new_publ_attrs2_len,
-                              curve_type);
+                              curve->curve_type);
```
Comment (ifranzki) [inline usr/lib/ep11_stdll/ep11_specific.c:5899]: and here

File: usr/lib/ep11_stdll/ep11_specific.c [hunk 5]
```diff
@@ -5926,7 +5906,7 @@ static CK_RV rsa_ec_generate_keypair(STDLL_TokData_t * tokdata,
     rc = check_key_attributes(tokdata, ktype, CKO_PRIVATE_KEY,
                               new_priv_attrs, new_priv_attrs_len,
                               &new_priv_attrs2, &new_priv_attrs2_len,
-                              curve_type);
+                              curve->curve_type);
```
Comment (ifranzki) [inline usr/lib/ep11_stdll/ep11_specific.c:5909]: and here

File: testcases/crypto/ec_func.c (hunk 1 above)
Comment (jschmidb) [reply testcases/crypto/ec_func.c]: Done. Routine rsa_ec_generate_keypair now checks if the curve is supported, before calling the ep11lib, and returns CKR_CURVE_NOT_SUPPORTED if not. This even makes local variable curve_type and local subroutine get_curve_type_from_attrs superfluous.

**Review by ifranzki — COMMENTED** [review]
Please also change the title of commit "TESTCASES: Fix known problem for pre-z15 processors", this is no longer a TESTCASE fix.

File: usr/lib/ep11_stdll/ep11_specific.c (hunk 5 above)
Comment (jschmidb) [reply usr/lib/ep11_stdll/ep11_specific.c]: Yes that's true ... done.

**Comment by jschmidb** [comment]
> Please also change the title of commit "TESTCASES: Fix known problem for pre-z15 processors", this is no longer a TESTCASE fix.

Done.

**Review by p-steuer — APPROVED** [review]

**Review by ifranzki — APPROVED** [review]

**Pull request merged**